### PR TITLE
[Search] clean up search api reports

### PR DESCRIPTION
**Pull request by emmaindal**
## Hey, I just made a Pull Request!

This PR covers api report clean up of the following packages:

* `@backstage/plugin-search-common`
* `@backstage/plugin-search-backend`
* `@backstage/plugin-search-backend-node`
* `@backstage/plugin-search-backend-module-elasticsearch`

It also [replaces all usages of @backstage/search-common with @backstage/plugin-search-common](https://github.com/backstage/backstage/pull/11805/commits/73fa59f6b20d34fd183ca6dc06a2c838267f1c84) as `@backstage/search-common` is moved there.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by iamEAP — APPROVED** [review]
✨ ✨ ✨

File: plugins/search-backend-module-elasticsearch/api-report.md [hunk 1]
```diff
@@ -15,17 +15,44 @@ import { Logger } from 'winston';
 import { SearchEngine } from '@backstage/plugin-search-common';
 import { SearchQuery } from '@backstage/plugin-search-common';
 
-// Warning: (ae-missing-release-tag) "ElasticSearchClientOptions" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
-// Warning: (ae-unresolved-link) The @link reference could not be resolved: The package "@backstage/plugin-search-backend-module-elasticsearch" does not have an export "ElasticSearchEngine"
-//
+// @public
+export type ConcreteElasticSearchQuery = {
```
Comment (Rugvip) [inline plugins/search-backend-module-elasticsearch/api-report.md:19]: Feeling that everything else is so nicely prefixed, perhaps make this `ElasticSearchConcreteQuery`?

**Review by Rugvip — APPROVED** [review]
Nice! ✨ 👍 

Side note just in case it's not know: when cleaning up API exports it's always best to take another look at the naming of symbols when weren't being exported. It's not a breaking change to change the name at that point, but it's the last opportunity :grin: It's somewhat common that the naming wasn't considered fully up until this point as well, since the types weren't exported anyway.

File: plugins/search-backend-module-elasticsearch/api-report.md (hunk 1 above)
Comment (emmaindal) [reply plugins/search-backend-module-elasticsearch/api-report.md:19]: Oh thats a great catch, I'll update that one! thanks @Rugvip

Comment (iamEAP) [reply plugins/search-backend-module-elasticsearch/api-report.md]: LGTM, @emmaindal!  :shipit:

**Pull request merged**